tests/app/views/test_drafts.py

```diff
@@ -808,33 +886,9 @@ def test_service_id_collisions_should_be_handled(self,
         res = self.publish_new_draft_service()
         assert_equal(res.status_code, 200)
 
-        with self.app.app_context():
-            # Count is 3 because we create on in the setup
-            assert_equal(Service.query.count(), 3)
-            assert_equal(DraftService.query.count(), 0)
-
-    @mock.patch('app.models.generate_new_service_id')
-    def test_draft_service_should_be_left_on_service_id_collision_failure(
-            self, generate_new_service_id):
-        generate_new_service_id.side_effect = [
-            '1234567890123457',
-            '1234567890123457',
-            '1234567890123457',
-            '1234567890123457',
-            '1234567890123457',
-            '1234567890123457',
-            '1234567890123457',
-        ]
-
-        res = self.publish_new_draft_service()
-        assert_equal(res.status_code, 200)
-        with assert_raises(IntegrityError):
-            res = self.publish_new_draft_service()
-
-        with self.app.app_context():
-            db.session.rollback()
-            assert_equal(Service.query.count(), 2)
-            assert_equal(DraftService.query.count(), 1)
+        # Count is 3 because we create one in the setup
+        assert self.service_count() == 3
```

Comment:
Is it also worth testing that the assigned service IDs are the two that we expect from the array?